### PR TITLE
Buggy-api-assignment

### PR DESCRIPTION
**Pull request by sahadevbite**
changes added for login from both user and not to save duplicate record.

### CODE REVIEW

**Review by parikshitd-webonise — CHANGES_REQUESTED** [review]
The code doesn't restrict a user from inserting duplicates records. The issue still persists on a database lavel.

File: src/main/java/com/webonise/login/controller/SignUpController.java [hunk 1]
```diff
@@ -18,6 +19,17 @@
 
     @PostMapping
     public UserDTO saveUser(@RequestBody UserRequest userRequest) {
-        return userService.saveUser(userRequest);
+    	int check = userService.checkUser(userRequest.getLoginId());
+    	System.err.println(check);
+    	if(check>0) {
+    		
+    		return null;
```
Comment (parikshitd-webonise) [inline src/main/java/com/webonise/login/controller/SignUpController.java]: Bad practice to return null.

File: src/main/java/com/webonise/login/controller/SignUpController.java [hunk 2]
```diff
@@ -18,6 +19,17 @@
 
     @PostMapping
     public UserDTO saveUser(@RequestBody UserRequest userRequest) {
-        return userService.saveUser(userRequest);
+    	int check = userService.checkUser(userRequest.getLoginId());
+    	System.err.println(check);
```
Comment (parikshitd-webonise) [inline src/main/java/com/webonise/login/controller/SignUpController.java]: We should log instead of `println`.

Comment (parikshitd-webonise) [inline src/main/java/com/webonise/login/controller/SignUpController.java]: We should we log.

File: src/main/java/com/webonise/login/controller/SignUpController.java [hunk 3]
```diff
@@ -1,23 +1,40 @@
...
+import com.webonise.login.model.UserDTO;
+import com.webonise.login.model.UserEntity;
+import com.webonise.login.model.UserRequest;
+import com.webonise.login.service.UserService;
+
 @RestController("/user")
 @RequestMapping("/")
 public class SignUpController {
+	
+	static Log logger = LogFactory.getLog(SignUpController.class);
```
Comment (sudharani-webonise) [inline src/main/java/com/webonise/login/controller/SignUpController.java:20]: why not private ?

File: src/main/java/com/webonise/login/controller/SignUpController.java [hunk 4]
```diff
@@ -1,23 +1,40 @@
...
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import com.webonise.login.model.UserDTO;
+import com.webonise.login.model.UserEntity;
+import com.webonise.login.model.UserRequest;
+import com.webonise.login.service.UserService;
+
 @RestController("/user")
 @RequestMapping("/")
```
Comment (sudharani-webonise) [inline src/main/java/com/webonise/login/controller/SignUpController.java:17]: what is the difference between @RestController, @RequestMapping what if I want to use only one Annotation

File: src/main/java/com/webonise/login/controller/SignUpController.java [hunk 5]
```diff
@@ -1,23 +1,40 @@
...
+	
+	static Log logger = LogFactory.getLog(SignUpController.class);
 
     @Autowired
     private UserService userService;
 
     @PostMapping
     public UserDTO saveUser(@RequestBody UserRequest userRequest) {
-        return userService.saveUser(userRequest);
+    	int check = userService.checkUser(userRequest.getLoginId());
```
Comment (sudharani-webonise) [inline src/main/java/com/webonise/login/controller/SignUpController.java:27]: include exception handling?

File: src/main/java/com/webonise/login/controller/SignUpController.java [hunk 6]
```diff
@@ -1,23 +1,40 @@
...
+    	logger.info(check);
+    	if(check>0) {
```
Comment (sudharani-webonise) [inline src/main/java/com/webonise/login/controller/SignUpController.java:29]: avoid magic number checks

**Pull request closed without merging**